GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                    {
+                        if (directoryObject is User)
+                        {
+                            Console.WriteLine("\n    " + ((IUser) (manager)).DisplayName);
+                        }
+                        else if (directoryObject is Contact)
+                        {
+                            Console.WriteLine("\n    " + ((IContact) (manager)).DisplayName);
+                        }
+
+                    }
+                    reports = await reports.GetNextPageAsync();
+                } while (reports != null);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting user's manager and reports {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            // group and role memberships
+            Console.WriteLine("\n Signed in user {0} is a member of the following Group and Roles (IDs)",
+                user.DisplayName);
+            IUserFetcher signedInUserFetcher = user;
+            try
+            {
+                IPagedCollection<IDirectoryObject> pagedCollection = await signedInUserFetcher.MemberOf.ExecuteAsync();
+                do
+                {
+                    List<IDirectoryObject> directoryObjects = pagedCollection.CurrentPage.ToList();
+                    foreach (IDirectoryObject directoryObject in directoryObjects)
+                    {
+                        if (directoryObject is Group)
+                        {
+                            Group group = directoryObject as Group;
+                            Console.WriteLine(" Group: {0}  Description: {1}", group.DisplayName, group.Description);
+                        }
+                        if (directoryObject is DirectoryRole)
+                        {
+                            DirectoryRole role = directoryObject as DirectoryRole;
+                            Console.WriteLine(" Role: {0}  Description: {1}", role.DisplayName, role.Description);
+                        }
+                    }
+                    pagedCollection = await pagedCollection.GetNextPageAsync();
+                } while (pagedCollection != null);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Signed in user's groups and roles memberships. {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            #endregion
+        }
+
+        private static async Task<User> CreateNewUser(IActiveDirectoryClient client, VerifiedDomain defaultDomain)
+        {
+            #region Create a new User
+
+            // **********************************************************
+            // Requires Directory.ReadWrite.All or Directory.AccessAsUser.All, and the signed in user
+            // must be a privileged user (like a company or user admin)
```

Comment:
Can we check if signed in user has this role? if he doesnt then we can display error saying "new user can not be created because of missing permissions"